cls/SourceControl/Git/Settings.cls

```diff
@@ -110,12 +113,16 @@ Method %Save() As %Status
     set @storage@("settings", "compileOnImport") = ..compileOnImport
     set @storage@("settings", "warnInstanceWideUncommitted") = ..warnInstanceWideUncommitted
     set @storage@("settings", "basicMode") = ..systemBasicMode
+    set @storage@("settings", "environmentName") = ..environmentName
     if ..basicMode = "system" {
         kill @storage@("settings", "user", $username, "basicMode")
     } else {
         set @storage@("settings", "user", $username, "basicMode") = ..basicMode
     } 
 
+    do $SYSTEM.Version.SystemMode(..environmentName)
```

Comment:
Please specifically test that someone with just %Developer can set this

Reply:
$SYSTEM.Version.SystemMode does not work for someone with %Developer

Reply:
Should I revert to setting the global directly (through a private helper method)?

Reply:
Setting the global directly isn't going to work either - I'd recommend just wrapping the whole thing in a Try/Catch and ignoring any errors.

Reply:
@isc-etamarch Yes, it'll just be a no-op.